src/Login.js

```diff
@@ -179,11 +192,62 @@ export async function sendLoginRequest(hsUrl, isUrl, loginType, loginParams) {
         }
     }
 
+    const dehydrationKeyCache = getDehydrationKeyCache();
+
     return {
         homeserverUrl: hsUrl,
         identityServerUrl: isUrl,
         userId: data.user_id,
         deviceId: data.device_id,
         accessToken: data.access_token,
+        rehydrationKeyInfo: dehydrationKeyCache.keyInfo,
+        rehydrationKey: dehydrationKeyCache.key,
+        olmAccount: data._olm_account,
+    };
+}
+
+async function getDehydrationKey(keyInfo) {
```

Comment:
Assuming this is actually what we want, let's attempt some de-duplication with the very similar code in `SecurityManager`.